cmd/kubectl-gadget/deploy.go

```diff
@@ -336,17 +354,25 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("it's not possible to use --quiet and --debug together")
 	}
 
-	objects, err := parseK8sYaml(resources.GadgetDeployment)
-	if err != nil {
-		return err
+	if !useSeccompProfile {
+		resourcesYAML[resourcesSeccompIndex] = ""
```

Comment:
nit: to avoid the `if` in the loop
```suggestion
		resourcesYAML = append(resourcesYAML[:resourcesSeccompIndex], resourcesYAML[resourcesSeccompIndex+1:]...)
```

Reply:
_I didn't read the resolved conversations. Maybe i should do that._

Performance wise it shouldn't matter as you stated. And this is not a hot path anyway.
My nit was more of a discussion question just to see what is the most go-idiomatic way to do this.

Would be fine either way